pkgs/development/mobile/androidenv/deploy-androidpackages.nix

```diff
@@ -0,0 +1,58 @@
+{stdenv, lib, unzip, mkLicenses}:
+{packages, os ? null, buildInputs ? [], patchesInstructions ? {}, meta ? {}, ...}@args:
+
+let
+  extraParams = removeAttrs args [ "packages" "os" "buildInputs" "patchesInstructions" ];
+  sortedPackages = builtins.sort (x: y: builtins.lessThan x.name y.name) packages;
+in
+stdenv.mkDerivation ({
+  pname = lib.concatMapStringsSep "-" (package: package.name) sortedPackages;
+  version = lib.concatMapStringsSep "-" (package: package.revision) sortedPackages;
+  src = map (package:
+    if os != null && builtins.hasAttr os package.archives then package.archives.${os} else package.archives.all
+  ) packages;
+  buildInputs = [ unzip ] ++ buildInputs;
```

Comment:
unzip goes to nativeBuildInputs

Reply:
do you mean something like this?
```
diff --git a/pkgs/development/mobile/androidenv/deploy-androidpackages.nix b/pkgs/development/mobile/androidenv/deploy-androidpackages.nix
index 96eebe2bc95..ff38e241bff 100644
--- a/pkgs/development/mobile/androidenv/deploy-androidpackages.nix
+++ b/pkgs/development/mobile/androidenv/deploy-androidpackages.nix
@@ -6,12 +6,13 @@ let
   sortedPackages = builtins.sort (x: y: builtins.lessThan x.name y.name) packages;
 in
 stdenv.mkDerivation ({
+  inherit buildInputs;
   pname = lib.concatMapStringsSep "-" (package: package.name) sortedPackages;
   version = lib.concatMapStringsSep "-" (package: package.revision) sortedPackages;
   src = map (package:
     if os != null && builtins.hasAttr os package.archives then package.archives.${os} else package.archives.all
   ) packages;
-  buildInputs = [ unzip ] ++ buildInputs;
+  nativeBuildInputs = [ unzip ];
   preferLocalBuild = true;

   unpackPhase = ''
```

it's not working! it gives me
```
unpacking source archive /nix/store/vhz1z6l82145m206kxh9m0k60cncr9ij-platform-tools_r33.0.3-linux.zip
do not know how to unpack source archive /nix/store/vhz1z6l82145m206kxh9m0k60cncr9ij-platform-tools_r33.0.3-linux.zip
```

Reply:
@Artturin can you help out why splicing seems to not work here?

Reply:
it's not due to splicing, nativeBuildInputs is being overriden by the `// extraParams`
https://github.com/NixOS/nixpkgs/blob/39bc9634f64232e817ddc910f5b683c146f2b454/pkgs/development/mobile/androidenv/platform-tools.nix#L5